src/reactor.cc

```diff
@@ -0,0 +1,332 @@
+    cyclus::toolkit::ResBuf<cyclus::Material>& inventory) {
+  if (!inventory.empty()) {
+    cyclus::Material::Ptr mat = inventory.Pop();
+    mat->Decay(context()->time());
+    inventory.Push(mat);
+  }
+}
+
+void Reactor::CombineInventory(
+    cyclus::toolkit::ResBuf<cyclus::Material>& inventory) {
+  if (!inventory.empty()) {
+    cyclus::Material::Ptr base = inventory.Pop();
+    int count = inventory.count();
+    for (int i = 0; i < count; i++) {
+      cyclus::Material::Ptr m = inventory.Pop();
+      base->Absorb(m);
+    }
+
+    inventory.Push(base);
+  }
+}
+
+void Reactor::ExtractHelium(
+    cyclus::toolkit::ResBuf<cyclus::Material>& inventory) {
+  if (!inventory.empty()) {
+    cyclus::Material::Ptr mat = inventory.Pop();
+    cyclus::toolkit::MatQuery mq(mat);
+    
+    cyclus::Material::Ptr helium = mat->ExtractComp(mq.mass(He3_id), He3_comp);
+
+    helium_storage.Push(helium);
+    inventory.Push(mat);
+  }
+}
+
+void Reactor::RecordEvent(std::string name, std::string val) {
+  context()
+      ->NewDatum("ReactorEvents")
+      ->AddVal("AgentId", id())
+      ->AddVal("Time", context()->time())
+      ->AddVal("Event", name)
+      ->AddVal("Value", val)
+      ->Record();
+}
+
+void Reactor::RecordOperationalInfo(std::string name, std::string val) {
+  context()
+      ->NewDatum("ReactorOperationsLog")
+      ->AddVal("AgentId", id())
+      ->AddVal("Time", context()->time())
+      ->AddVal("Event", name)
+      ->AddVal("Value", val)
+      ->Record();
+}
+
+void Reactor::RecordStatus(std::string status, double power) {
+  context()
+      ->NewDatum("ReactorStatus")
+      ->AddVal("AgentId", id())
+      ->AddVal("Time", context()->time())
+      ->AddVal("Status", status)
+      ->AddVal("Power", power)
+      ->Record();
+}
+
+void Reactor::RecordInventories(double storage, double excess, double sequestered, 
+                            double blanket, double blanket_excess, double helium) {
+  context()
+      ->NewDatum("ReactorInventories")
+      ->AddVal("AgentId", id())
+      ->AddVal("Time", context()->time())
+      ->AddVal("TritiumStorage", storage)
+      ->AddVal("TritiumExcess", excess)
+      ->AddVal("TritiumSequestered", sequestered)
+      ->AddVal("LithiumBlanket", blanket)
+      ->AddVal("BlanketExcess", blanket_excess)
+      ->AddVal("HeliumStorage", helium)
+      ->Record();
+}
+
+void Reactor::DepleteBlanket(double bred_tritium_atoms) {
+  cyclus::Material::Ptr blanket_mat = blanket.Pop();
+  cyclus::toolkit::MatQuery b(blanket_mat);
+  cyclus::CompMap depleted_comp;
+
+  double converted_Li6 = Li6_contribution * bred_tritium_atoms;
+  double converted_Li7 = Li7_contribution * bred_tritium_atoms;
+  double bred_He4 = bred_tritium_atoms;
+  double blanket_Li6 = b.moles(Li6_id)*avagadros_number;
+  double blanket_Li7 = b.moles(Li7_id)*avagadros_number;
+  double blanket_tritium = b.moles(tritium_id)*avagadros_number;
+  double blanket_He4 = b.moles(He4_id)*avagadros_number;
```

Comment:
and now I wonder if you can do this on a mole basis instead of atoms....

Reply:
Combined those variables, and yes. Yes you can do this on a mole basis. It cleans up some of the multiplying by Avagadro's number (though ultimately you do still have to multiply/divide by it once or twice since at the end of the day you need mass).